Public job character infos

herpaderpaldent opened the pull request:
This update allows dispatch public jobs without a refresh_token

This job is needed to cleanup character_info entries without a valid refresh_token anymore. 

E.g. A character is leaving the corporation or is about to awox the corporation and invalidates his refresh_tokens. Eventhough the character might have left long time ago. The character still appears f.e. as in corporation or member_tracking views. This will update character informations that are publicly available.

leonjza commented on src/Jobs/EsiBase.php:
```diff
@@ -179,12 +179,15 @@ public function getCharacterRoles(): array
      * @throws \Exception
      */
-    public function getCharacterId(): int
+    public function getCharacterId(): ?int
```
I don't think we should loosen the type here. This change means we suddenly have to add `is_null`-type checks everywhere this is used which is complexity++ in a word where refresh tokens are almost always expected.

leonjza commented on src/Jobs/Character/Info.php:
```diff
@@ -63,24 +64,33 @@ public function handle()
-        $character_info = $this->retrieve([
-            'character_id' => $this->getCharacterId(),
-        ]);
+        $character_ids = ! is_null($this->getCharacterId()) ? collect($this->getCharacterId()) :
```
I think instead of monkey patching the authenticated job, lets create a new one called `PublicInfo` [here](https://github.com/eveseat/eveapi/tree/master/src/Jobs/Character) that explicitly handle's characters that don't have refresh tokens.

That being said, if we are going to add this, we need to make it clear in the character view that a specific character that is being viewed is a public profile (aka: we don't have a valid token for this character).

herpaderpaldent replied:
regarding:

> we need to make it clear in the character view that a specific character that is being viewed is a public profile (aka: we don't have a valid token for this character).

i added the following PR: https://github.com/eveseat/web/pull/273

I did not link these PR as i see them independently.

herpaderpaldent commented:
Hey @leonjza,
Thank you for your valuable input. I just reverted the changes introduced in `EsiBase` instead i introduced a `PublicInfo` job which extends `Info`. Furthermore i refactored the `Info` job to be more generic and to reuse its public methods from a child class.

i too confirmed that the public tag is attached to the job:
![image](https://user-images.githubusercontent.com/6583519/51202277-ac1f4100-18fe-11e9-9588-b0a8f6bf5c1c.png)

leonjza commented:
I'd prefer we not couple jobs and keep them isolated, please.

leonjza commented:
Otherwise, this will definitely work better I rate. Thanks!

herpaderpaldent commented:
Ok i adapted the PR to the findings we had on slack.
